### PR TITLE
[Syncope-808] Add netbeans plugin to Apache Syncope

### PR DESCRIPTION
**Pull request by NuwanSameera**

### CODE REVIEW

File: common/lib/hs_err_pid3976.log [hunk 1]
```diff
@@ -0,0 +1,370 @@
+#
```
Comment (andrea-patricelli) [inline common/lib/hs_err_pid3976.log]: Please remove this file

File: netbeans-plugin/nb-configuration.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (andrea-patricelli) [inline netbeans-plugin/nb-configuration.xml]: Missing license header

File: netbeans-plugin/optin.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<html> <body> <h3>Hi ${user.plainAttrMap["firstname"].values[0]} ${user.plainAttrMap["surname"].values[0]}, welcome to Syncope!</h3>  <p>    Your username is ${user.username}.<br/>    Your email address is ${user.plainAttrMap["email"].values[0]}.    Your email address inside a <a href="http://localhost/?email=${user.plainAttrMap["email"].values[0].replace('@', '%40')}">link</a>. </p>  <p>     This message was sent to the following recipients: <ul>
```
Comment (andrea-patricelli) [inline netbeans-plugin/optin.html]: Remove this file, it is unseful. You have to take templates from Syncope.

File: netbeans-plugin/pom.xml [hunk 4]
```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (andrea-patricelli) [inline netbeans-plugin/pom.xml]: Missing license header (please insert  the correct License header!!). Also plugin versions must be specified into parent pom.xml and then used into child pom.xml

File: netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/*
```
Comment (andrea-patricelli) [inline netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java]: Wrong license header

File: netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/view/ServerDetailsView.form [hunk 6]
```diff
@@ -0,0 +1,144 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (andrea-patricelli) [inline netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/view/ServerDetailsView.form]: Missing license header

File: netbeans-plugin/src/main/nbm/manifest.mf [hunk 7]
```diff
@@ -0,0 +1,4 @@
+Manifest-Version: 1.0
```
Comment (andrea-patricelli) [inline netbeans-plugin/src/main/nbm/manifest.mf]: Manifest must be generated with maven plugin automatically

File: netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/bundle.properties [hunk 8]
```diff
@@ -0,0 +1,6 @@
+#Localized module labels. Defaults taken from POM (<name>, <description>, <groupId>) if unset.
```
Comment (andrea-patricelli) [inline netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/bundle.properties]: All properties are commented, is this file really useful?

File: netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/view/Bundle.properties [hunk 9]
```diff
@@ -0,0 +1,49 @@
+TemplateViewTopComponent.jPanel1.TabConstraints.tabTitle=TEXT
```
Comment (andrea-patricelli) [inline netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/view/Bundle.properties]: random license appearances, why?

**Comment by andrea-patricelli** [comment]
Hi Nuwan,
see comments above, there is my feedback about PR.

- Pay attention to license headers, they are wrong! You can take correct license from whatever file into the project (for example root pom.xml).
- Can you also rename PR by including tag [SYNCOPE-808]?

File: common/lib/hs_err_pid3976.log (hunk 1 above)
Comment (NuwanSameera) [inline common/lib/hs_err_pid3976.log]: Remove file

File: netbeans-plugin/nb-configuration.xml (hunk 2 above)
Comment (NuwanSameera) [inline netbeans-plugin/nb-configuration.xml]: License header added. This file generate by netbeans. Project doesn't build without this file.

File: netbeans-plugin/optin.html (hunk 3 above)
Comment (NuwanSameera) [inline netbeans-plugin/optin.html]: It is removed.

File: netbeans-plugin/pom.xml (hunk 4 above)
Comment (NuwanSameera) [inline netbeans-plugin/pom.xml]: Added lenience header.

File: netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java (hunk 5 above)
Comment (NuwanSameera) [inline netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java]: Change lenience header.

File: netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/view/ServerDetailsView.form (hunk 6 above)
Comment (NuwanSameera) [inline netbeans-plugin/src/main/java/org/apache/syncope/netbeans/plugin/view/ServerDetailsView.form]: Add license header to .form files.

File: netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/bundle.properties (hunk 8 above)
Comment (NuwanSameera) [inline netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/bundle.properties]: Remove org/apache/syncope/netbeans/plugin/bundle.properties file.

File: netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/view/Bundle.properties (hunk 9 above)
Comment (NuwanSameera) [inline netbeans-plugin/src/main/resources/org/apache/syncope/netbeans/plugin/view/Bundle.properties]: Remove random license and add correct license header.

File: netbeans-plugin/src/main/nbm/manifest.mf (hunk 7 above)
Comment (NuwanSameera) [inline netbeans-plugin/src/main/nbm/manifest.mf]: This file generated by netbeans automatically and it edit by netbeans. Project don't build without it.

**Comment by NuwanSameera** [comment]
Removed 
<properties>
       <checkstyle.skip>true</checkstyle.skip>
       <rat.skip>true</rat.skip>
   </properties>

project have to build with checkstyle and rat checks enabled

**Comment by NuwanSameera** [comment]
Solved..
 [WARNING] The artifact org.apache.commons:commons-io:jar:1.3.2 has been relocated to commons-io:commons-io:jar:1.3.2

**Comment by NuwanSameera** [comment]
You can save template remotely using
   Right click : Apache Syncope in tree and select Save. 
It save last editing source. This need to improve.

**Comment by NuwanSameera** [comment]
Add progress bar to plugin to show template loading.

**Comment by ilgrosso** [comment]
Please rebase this PR and move the changes under `ide/netbeans` (similarly to the now existing `ide/eclipse`)

**Pull request closed without merging**